packages/debugger/src/handlers/editor.ts

```diff
@@ -277,6 +278,36 @@ export class EditorHandler implements IDisposable {
     );
   }
 
+  /**
+   * Handle a click on the gutter.
+   *
+   * @param editor The editor from where the click originated.
+   * @param position The position corresponding to the click event.
+   */
+  private _getEffectiveClickedLine(
+    editor: EditorView,
+    position: number
+  ): [Line | undefined, boolean] {
+    let clickedLine = editor.state.doc.lineAt(position);
+    let clickedLineNumber = clickedLine.number;
+    let targetLine: Line | undefined = undefined;
+    let isLineEmpty: boolean =
+      false; /* is true is the clicked line of code is empty */
+    if (clickedLine.text.trim() === '') {
+      isLineEmpty = true;
+      while (clickedLineNumber > 1) {
+        clickedLineNumber--;
+        const prevLine = editor.state.doc.line(clickedLineNumber);
+        if (prevLine.text.trim() !== '') {
+          targetLine = prevLine;
+          break;
+        }
+      }
+    } else {
+      (isLineEmpty = false), (targetLine = clickedLine);
```

Comment:
This is rather unusual style. Is this intentional?

Reply:
No it is not intentional. Will change it to a more usual style.